tests/Phinx/Db/Adapter/SQLiteAdapterTest.php

```diff
@@ -1371,6 +1372,208 @@ public function testAlterTableWithConstraints()
         }
     }
 
+    /**
+     * Tests that operations that trigger implicit table drops will not cause
+     * a foreign key constraint violation error.
+     */
+    public function testAlterTableDoesNotViolateRestrictedForeignKeyConstraint()
+    {
+        $this->adapter->execute('PRAGMA foreign_keys = ON');
+
+        $articlesTable = new \Phinx\Db\Table('articles', [], $this->adapter);
+        $articlesTable
+            ->insert(['id' => 1])
+            ->save();
+
+        $commentsTable = new \Phinx\Db\Table('comments', [], $this->adapter);
+        $commentsTable
+            ->addColumn('article_id', 'integer')
+            ->addForeignKey('article_id', 'articles', 'id', [
+                'update' => ForeignKey::RESTRICT,
+                'delete' => ForeignKey::RESTRICT,
+            ])
+            ->insert(['id' => 1, 'article_id' => 1])
+            ->save();
+
+        $this->assertTrue($this->adapter->hasForeignKey('comments', ['article_id']));
+
+        $articlesTable
+            ->addColumn('new_column', 'integer')
+            ->update();
+
+        $articlesTable
+            ->renameColumn('new_column', 'new_column_renamed')
+            ->update();
+
+        $articlesTable
+            ->changeColumn('new_column_renamed', 'integer', [
+                'default' => 1,
+            ])
+            ->update();
+
+        $articlesTable
+            ->removeColumn('new_column_renamed')
+            ->update();
+
+        $articlesTable
+            ->addIndex('id', ['name' => 'ID_IDX'])
+            ->update();
+
+        $articlesTable
+            ->removeIndex('id')
+            ->update();
+
+        $articlesTable
+            ->addForeignKey('id', 'comments', 'id')
+            ->update();
+
+        $articlesTable
+            ->dropForeignKey('id')
+            ->update();
+
+        $articlesTable
+            ->addColumn('id2', 'integer')
+            ->addIndex('id', ['unique' => true])
+            ->changePrimaryKey('id2')
+            ->update();
+    }
+
+    /**
+     * Tests that foreign key constraint violations introduced around the table
+     * alteration process (being it implicitly by the process itself or by the user)
+     * will trigger an error accordingly.
+     */
+    public function testAlterTableDoesViolateForeignKeyConstraintOnTargetTableChange()
+    {
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage('Integrity constraint violation: FOREIGN KEY constraint on `comments` failed.');
```

Comment:
Can we put this right above the line that'll trigger it? That way, we won't risk some other line unexpectedly throwing this error, and miss the regression.

Reply:
Sure thing, its done.